test/notebooks/notebook_test_utils.py

```diff
@@ -0,0 +1,140 @@
+
+
+def _init_secret_store(secrets: Secrets) -> None:
+    secrets.save(CKey.use_itde, 'yes')
+    secrets.save(CKey.mem_size, '4')
+    secrets.save(CKey.disk_size, '4')
+    secrets.save(CKey.db_schema, 'NOTEBOOK_TESTS')
+
+
+def _insert_hacks(nb: nbformat.NotebookNode, hacks: List[Tuple[str, str]]):
+
+    def cell_match(nb_cell, ins_tag: str) -> bool:
+        return ('tags' in nb_cell.metadata) and (ins_tag in nb_cell.metadata['tags'])
+
+    for hack in hacks:
+        insertion_tag, hack_content = hack
+        # Find the sequential numbers of the target cells.
+        location_cell_nums = [cell_no for cell_no, nb_cell in enumerate(nb.cells)
+                              if cell_match(nb_cell, insertion_tag)]
+        # Insert the hack cell after each cell located at the previous step.
+        for cell_no in location_cell_nums[::-1]:
+            hack_cell = nbformat.v4.new_code_cell(hack_content)
+            nb.cells.insert(cell_no + 1, hack_cell)
+
+
+def run_notebook(notebook_file: str, store_file: str, store_password: str,
+                 timeout: int = -1, hacks: Optional[List[Tuple[str, str]]] = None) -> None:
+    """
+    Executes notebook with added access to the configuration store.
+
+    Parameters:
+        notebook_file:  Notebook file.
+        store_file:     Configuration store file.
+        store_password: Configuration store password.
+        timeout:        Optional timeout in seconds
+        hacks:          Optional hacks to be inserted into the notebook
+                        before running it.
+    """
+
+    nb = nbformat.read(notebook_file, as_version=4)
+
+    if hacks:
+        _insert_hacks(nb, hacks)
+
+    # Insert the following code at the beginning of the notebook
+    init_code = f'''
+    def init_notebook_test():
+        from pathlib import Path
+        from exasol.secret_store import Secrets
+        global sb_config
+        sb_config = Secrets(Path("{store_file}"), "{store_password}")
+    init_notebook_test()
+    '''
+    nb.cells.insert(0, nbformat.v4.new_code_cell(init_code))
+
+    # Execute the notebook object, expecting to get no exceptions.
+    nb_client = NotebookClient(nb, timeout=timeout, kernel_name='python3')
+    nb_client.execute()
+
+
+@pytest.fixture
+def access_to_temp_secret_store(tmp_path: Path) -> Tuple[Path, str]:
+    """
+    Creates a temporary configuration store.
+    Brings up and subsequently destroys the Docker-DB.
+    Returns the temporary configuration store path and password.
+    """
+
+    # Create temporary secret store
+    store_path = tmp_path / 'tmp_config.sqlite'
+    # Should the password be constant or randomly generated is debatable.
+    # Strictly speaking, random password makes a test non-deterministic.
+    # On the other hand, it improves the security. If for some reason the
+    # temporary configuration store is not deleted, there will be no access
+    # to it since the password will be forgotten once the session is over.
```

Comment:
good question, in the CI and from the test setup, everything should be temporary, so nothing should survive for long. The main concerns could be the AWS credentials and other secrets. From that point, maybe it is actually a good idea, to use a random password.

I would also recommend to add finally block which removes the secret store file

Reply:
It should be handled by the pytest's tmp_path fixture.